contracts/HtsSystemContractJson.sol

```diff
@@ -441,6 +442,42 @@ contract HtsSystemContractJson is HtsSystemContract {
         return slot;
     }
 
+    function _tokenUriSlot(uint32 serialId) internal override virtual returns (bytes32) {
+        bytes32 slot = super._tokenUriSlot(serialId);
+        if (vm.load(_scratchAddr(), slot) == bytes32(0)) {
+            string memory metadata = mirrorNode().getNftMetadata(address(this), serialId);
+            string memory uri = string(decode(metadata));
+            storeString(address(this), uint256(slot), uri);
+        }
+        return slot;
+    }
+
+    function _ownerOfSlot(uint32 serialId) internal override virtual returns (bytes32) {
+        bytes32 slot = super._ownerOfSlot(serialId);
+        if (vm.load(_scratchAddr(), slot) == bytes32(0)) {
+            address owner = mirrorNode().getNftOwner(address(this), serialId);
+            _setValue(slot, bytes32(uint256(uint160(owner))));
+        }
+        return slot;
+    }
+
+    function _getApprovedSlot(uint32 serialId) internal override virtual returns (bytes32) {
+        bytes32 slot = super._getApprovedSlot(serialId);
+        if (vm.load(_scratchAddr(), slot) == bytes32(0)) {
+            address approved = mirrorNode().getNftSpender(address(this), serialId);
+            _setValue(slot, bytes32(uint256(uint160(approved))));
+        }
+        return slot;
+    }
+
+    function _isApprovedForAllSlot(address owner, address operator) internal override virtual returns (bytes32) {
+        bytes32 slot = super._isApprovedForAllSlot(owner, operator);
+        if (vm.load(_scratchAddr(), slot) == bytes32(0)) {
+            // TODO: Implement this
```

Comment:
@acuarica I'm not entirely sure how to retrieve this information from the Mirrornode, or if it's even possible. :smiling_face_with_tear: . I've been trying to deduce it from the documentation, but I couldn't find anything that matches the 'isApprovedForAll' behavior.

I’ve also been considering intermediary solution:
- Counting all NFTs assigned to a user and then checking how many of those tokens the spender is allowed to spend. If those numbers match, perhaps just returning `true` could work. However, counting them would require listing them first, which means retrieving all pages using the 'next' link...

That said, I’m not sure if this is a good approach, so I wanted to ask for your opinion first. What do you think?